service/controller/resource/workermigration/create.go

```diff
@@ -0,0 +1,62 @@
+package workermigration
+
+import (
+	"context"
+
+	"github.com/giantswarm/microerror"
+	expcapiv1alpha3 "sigs.k8s.io/cluster-api/exp/api/v1alpha3"
+
+	"github.com/giantswarm/azure-operator/v4/service/controller/key"
+	"github.com/giantswarm/azure-operator/v4/service/controller/resource/workermigration/internal/azure"
+)
+
+// EnsureCreated ensures that built-in workers are migrated to node pool.
+func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
+	cr, err := key.ToCustomResource(obj)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	r.logger.LogCtx(ctx, "level", "debug", "message", "ensuring that built-in workers are migrated to node pool")
+
+	var builtinVMSS azure.VMSS
+	{
+		builtinVMSS, err = r.azureapi.GetVMSS(ctx, key.ResourceGroupName(cr), key.WorkerVMSSName(cr))
+		if azure.IsNotFound(err) {
+			r.logger.LogCtx(ctx, "level", "debug", "message", "built-in workers don't exist anymore")
+			r.logger.LogCtx(ctx, "level", "debug", "message", "canceling resource")
+
+			return nil
+		} else if err != nil {
+			return microerror.Mask(err)
+		}
+	}
+
+	var machinePool expcapiv1alpha3.MachinePool
+	//var azureMachinePool expcapzv1alpha3.AzureMachinePool
+	{
+		// TODO: Ensure that MachinePool & AzureMachinePool CRs exist for builtinVMSS.
```

Comment:
The handler is still very much WIP due to my TDD approach. Currently there's a failing test for this so that I can quickly catch the context on Monday and continue.